src/PowerShellEditorServices/Utility/VisitorUtils.cs

```diff
@@ -10,42 +16,336 @@ namespace Microsoft.PowerShell.EditorServices.Utility
+        /// <returns>A tuple with start column and line for the function name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(FunctionDefinitionAst functionDefinitionAst)
+        {
+            int startOffset = functionDefinitionAst.IsFilter ? "filter".Length : functionDefinitionAst.IsWorkflow ? "workflow".Length : "function".Length;
+            return GetNameStartColumnAndLineFromAst(functionDefinitionAst, startOffset);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual class/enum name in a type definition AST.
+        /// </summary>
+        /// <param name="typeDefinitionAst">A TypeDefinitionAst object in the script's AST</param>
+        /// <returns>A tuple with start column and line for the type name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(TypeDefinitionAst typeDefinitionAst)
+        {
+            int startOffset = typeDefinitionAst.IsEnum ? "enum".Length : "class".Length;
+            return GetNameStartColumnAndLineFromAst(typeDefinitionAst, startOffset);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual method/constructor name in a function member AST.
+        /// </summary>
+        /// <param name="functionMemberAst">A FunctionMemberAst object in the script's AST</param>
+        /// <returns>A tuple with start column and line for the method/constructor name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(FunctionMemberAst functionMemberAst)
+        {
+            // find name index to get offset even with attributes, static, hidden ++
+            int nameStartIndex = functionMemberAst.Extent.Text.LastIndexOf(string.Concat(functionMemberAst.Name, '('), StringComparison.OrdinalIgnoreCase);
+            return GetNameStartColumnAndLineFromAst(functionMemberAst, nameStartIndex);
+        }
+
+        /// <summary>
+        /// Calculates the start line and column of the actual property name in a property member AST.
+        /// </summary>
+        /// <param name="propertyMemberAst">A PropertyMemberAst object in the script's AST</param>
+        /// <param name="isEnumMember">A bool indicating this is a enum member</param>
+        /// <returns>A tuple with start column and line for the property name</returns>
+        internal static (int startColumn, int startLine) GetNameStartColumnAndLineFromAst(PropertyMemberAst propertyMemberAst, bool isEnumMember)
+        {
+            // find name index to get offset even with attributes, static, hidden ++
+            string searchString = isEnumMember ? propertyMemberAst.Name : string.Concat('$', propertyMemberAst.Name);
+            int nameStartIndex = propertyMemberAst.Extent.Text.LastIndexOf(searchString, StringComparison.OrdinalIgnoreCase);
```

Comment:
Same here as `InitialValue` can have any arbitrary text

Reply:
Agree. Should've been `IndexOf()` also.

`$` aren't used in neither types or keywords like static, hidden etc. EnumMember can't have anything prefixed AFAIK, so shouldn't be an issue, right?